conans/test/integration/test_pkg_signing.py

```diff
@@ -0,0 +1,43 @@
+import os
+import textwrap
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+from conans.util.files import save
+
+
+def test_pkg_sign():
+    c = TestClient(default_server_user=True)
+    c.save({"conanfile.py": GenConanfile("pkg", "0.1").with_exports("export/*")
+            .with_exports_sources("export_sources/*").with_package_file("myfile", "mycontents!"),
+            "export/file1.txt": "file1!",
+            "export_sources/file2.txt": "file2!"})
+    signer = textwrap.dedent(r"""
+        import os
+
+        def sign(ref, files, folder):
+            print("Signing ref: ", ref)
+            print("Signing files: ", files)
+            signature = os.path.join(folder, "signature.asc")
+            open(signature, "w").write("\n".join(files))
+            return ["signature.asc"]
```

Comment:
This is a bit ugly, needing to return the name of the created files, but it is necessary to add the new files to the "upload bundle". We could try to auto-detect which files are new, but sounds trickier.

File paths are also absolute, I don't love it, but this is necessary because they can be in different folders, as ``conanfile.py`` and ``conanmanifest.txt`` for example might be in the ``export`` folder, while the ``conan_sources.tgz`` is created directly in the ``download`` folder, and the 3 files are the final package. It would be more elegant if just one folder of all files is passed, but for that we would need to copy ``conanfile.py`` and manifest from export->download. Is doable, but not what we are doing now.

Reply:
I'm ok with the return, as far as it is documented sounds reasonable.
About the absolute paths, the same, is similar to the `cpp_info` fields, having the absolute path is very convenient.